EGO Forum Enhancement.user.js

```diff
@@ -0,0 +1,301 @@
+// ==UserScript==
+// @name         EdgeGamers Forum Enhancement
+// @namespace    https://github.com/MSWS/eGOMonkey
+// @version      3.0.0
+// @description  Add various enhancements & QOL additions to the EdgeGamers Forums that are beneficial for Leadership members.
+// @author       blank_dvth, Skle, MSWS
+// @match        https://www.edgegamers.com/*
+// @icon         https://www.google.com/s2/favicons?sz=64&domain=edgegamers.com
+// @require      https://peterolson.github.io/BigInteger.js/BigInteger.min.js
+// @require      https://raw.githubusercontent.com/12pt/steamid-converter/master/js/converter-min.js
+// @grant        none
+// ==/UserScript==
+
+'use strict';
+const MAUL_BUTTON_TEXT = "MAUL";
+
+/**
+ * Creates a button and adds it to the given div
+ * @param {string} href URL that button should link to
+ * @param {string} text Buttons' text
+ * @param {HTMLDivElement} div Div to add/append to
+ * @param {string} target Meta target for button
+ * @param {boolean} append True to append, false to insert
+ */
+function createButton(href, text, div, target = "_blank", append = false) {
+    var button = document.createElement("a");
+    button.href = href;
+    button.target = target;
+    button.classList.add('button--link', 'button');
+
+    var button_text = document.createElement("span"); // Create button text
+    button_text.classList.add('button-text');
+    button_text.innerHTML = text;
+
+    // Add all elements to their respective parents
+    button.appendChild(button_text);
+    append ? div.appendChild(button, div.lastElementChild) : div.insertBefore(button, div.lastElementChild);
+}
+
+/**
+ * Adds a MAUL profile button to the given div
+ * @param {HTMLDivElement} div Div to add to
+ * @param {number} member_id Member's ID
+ */
+function addMAULProfileButton(div, member_id) { createButton("https://maul.edgegamers.com/index.php?page=home&id=" + member_id, MAUL_BUTTON_TEXT, div); }
+
+/**
+ * Adds a "List Bans" button to the div
+ * @param {HTMLDivElement} div Div to add to
+ * @param {number} steam_id_64 Steam ID to check
+ * TODO: Add support for other game IDs
+ */
+function addBansButton(div, steam_id_64) { createButton("https://maul.edgegamers.com/index.php?page=bans&qType=gameId&q=" + steam_id_64, "List Bans", div); }
+
+/**
+ * Adds a "Lookup ID" button to the div
+ * @param {HTMLDivElement} div Div to add to
+ * @param {number} post_title Steam ID to lookup
+ */
+function addLookupButton(div, post_title) {
+    var steam_id_unknown = post_title.match(/^.* - .* - (?<game_id>[\w\d\/\[\]\-\.:]*)$/);
+    if (steam_id_unknown)
+        createButton("https://steamid.io/lookup/" + steam_id_unknown.groups.game_id, "Lookup ID", div);
+}
+
+/**
+ * Adds a "Move to Complted" button to the div
```

Comment:
`Completed` is misspelled. This comment may be redundant if the next one is implemented.